pkg/kubelet/cm/internal_container_lifecycle_test.go

```diff
@@ -0,0 +1,30 @@
+package cm
+
+import (
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/memorymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"testing"
+)
+
+func TestPreStartContainer(t *testing.T) {
```

Comment:
So, looking at the implementation for PreStartContainer we have a few things that we can test:

- That when a CPU manager is provided it has AddContainer called
- When a CPU manager is not provided, it's ok
- That when a Memory manager is provided it has AddContainer called
- When a Memory manager is not provided, it's ok
- That TopologyManager's AddContainer is only called when the feature flag is enabled

Reply:
@endocrimes  Thanks for helping . I have tried implementing this using the gomock lib. I mocked the Managers AddContainer method and  called PreStartContainer then checked if it was called or not.

Reply:
But I am facing issue . I don't know why it is failing. Can you suggest some change ?

Reply:
Hi @prakharporwal 

The reason it fails is that you are specifying `topologymanager.NewFakeManager()`  as the member of `i.topologyManager`.
Specify `mockTopologyManager` there and it worked fine.